tools/cli/main.c

```diff
@@ -205,13 +212,30 @@ TDNFCliPrintError(
         dwErrorCode = 0;
     }
 
-    if (dwErrorCode)
+    if (doJson)
     {
-        pr_err("Error(%u) : %s\n", dwErrorCode, pszError);
+        if (dwErrorCode)
+        {
+            struct json_dump *jd;
+            jd = jd_create(0);
+            jd_map_start(jd);
+            jd_map_add_int(jd, "Error", dwErrorCode);
+            jd_map_add_string(jd, "ErrorMessage", pszError);
+            
+            pr_json(jd->buf);
+            jd_destroy(jd);
+        }
     }
     else
     {
-        pr_err("%s\n", pszError);
+        if (dwErrorCode)
```

Comment:
```
        if (dwErrorCode)
        {
            pr_err("Error(%u) : ", dwErrorCode);
        }
        pr_err("%s\n", pszError);
```

Reply:
Thanks, but I don't like it. Makes it too obscure what's happening.